### DISCUSSION TITLE
Headless component combobox with @entangle value

### DISCUSSION THREAD

**Discussion by m15studio** [livewire/livewire#6606] [1. Help]
Hi there,
I'm using new alpinejs headless combobox inside a livewire component.
On livewire component I have $customer_id and I bind it with alpinejs value using @entangle('customer_id').live

When I click on combobox, instead to send to livewire only one single Ajax request with value, alpine component send continuously Ajax requests to update component property $customer_id

<img width="1026" alt="Screenshot 2023-08-31 alle 10 27 08" src="https://github.com/livewire/livewire/assets/123408679/e6cd2765-917b-4a84-afa6-aaaed3c3cb35">

<img width="240" alt="Screenshot 2023-08-31 alle 10 29 30" src="https://github.com/livewire/livewire/assets/123408679/ad1088f5-0aa9-4caa-a6ee-54ef0d85a753">

```
<div x-data="{
selected:  @entangle('customer_id').live,
frameworks: [
{
id: 1,
name: 'Laravel',
disabled: false,
},
{
id: 2,
name: 'Ruby on Rails',
disabled: false,
},
{
id: 3,
name: 'Phoenix',
disabled: true,
},
]}"></div>
```

If I use
`value:  @entangle('customer_id').live,`
inside Choices component (https://alpinejs.dev/component/choices) no-headless I haven't problems

**Comment by dwightwatson** [comment]
I'm having similar issues with the headless combobox and entangle.

I have used `query: @entable('query').live` (rather than the selected value) and it appears to re-set the query string on each update AND the text input loses focus

**Comment by bawbanksy** [comment]
Same issue here.

**Comment by cbaconnier** [comment]
Livewire version: 3.0.3

Using it `wire:ignore` works for me, but updating the value from livewire doesn't update correctly the display of the value in the combobox. I have to click outside of the combobox to update it. 
But this is an issue with the combobox, because it can be reproduced without Livewire.

```php
<?php
 
namespace App\Livewire;
 
use Livewire\Component;
use Livewire\Attributes\Url;
 
class Demo extends Component
{

    #[Url]
    public $selected;

    public function setLaravel()
    {
        $this->selected = $this->options[0];
    }

    public function setDjango()
    {
        $this->selected = $this->options[2];
    }

    public function setAdonis()
    {
        $this->selected = $this->options[5];
    }

    public $options = [
        [
            'id' => 1,
            'name' => 'Laravel',
            'disabled' => false,
        ],
        [
            'id' => 2,
            'name' => 'Ruby on Rails',
            'disabled' => false,
        ],
        [
            'id' => 3,
            'name' => 'Django',
            'disabled' => false,
        ],
        [
            'id' => 4,
            'name' => 'Express',
            'disabled' => true,
        ],
        [
            'id' => 5,
            'name' => 'Phoenix',
            'disabled' => false,
        ],
        [
            'id' => 6,
            'name' => 'Adonis',
            'disabled' => false,
        ],
        [
            'id' => 7,
            'name' => 'NextJS',
            'disabled' => false,
        ],
    ];
 
    public function render()
    {
        return view('livewire.demo');
    }
}
```

```blade
<x-select wire:model.live="selected" :options="$this->options" />
```


```blade
<div
    wire:ignore
    x-data="{
        query: '',
        selected: @entangle($attributes->wire('model')),
        frameworks: @js($options),
        get filteredFrameworks() {
            return this.query === ''
                ? this.frameworks
                : this.frameworks.filter((framework) => {
                    return framework.name.toLowerCase().includes(this.query.toLowerCase())
                })
        },
    }"
    class="max-w-xs w-full"
>
  {{-- skipped --}}
    <button  class="absolute inset-y-0 right-0 flex items-center mr-10" x-on:click="selected = ''">x</button>
  {{-- skipped --}}
</div>
```


https://github.com/livewire/livewire/assets/4738184/52cc3bfc-6c0a-4af2-a9e1-e2fb581c0516

**Reply by cbaconnier** [reply]
Worth mentionning: I have also mention the issue I described here on Alpine discussion: https://github.com/alpinejs/alpine/discussions/3724 Because I don't think it's a Livewire issue.

- I'm able update the select input when I clear the field by using `$refs.__input.value = ''` ref: https://github.com/alpinejs/alpine/discussions/3724#discussioncomment-7003302
- I'm pretty sure the issue can be worked around by using `$watch` on `selected` to update the  input value `$refs.__input.value = ...`

But I believe the behavior should be changed in Alpine UI Component itself to allow update from outside of the component.

**Comment by bawbanksy** [comment]
@m15studio Have you had any further success? Im still chipping away at this even running through the beta docs, but not making headway.

**Reply by m15studio** [reply]
@bawbanksy no, for now I'm using Choices component
Have you try with the @cbaconnier's solution?

**Comment by gdebrauwer** [comment]
This should be fixed: https://github.com/livewire/livewire/pull/7776
